octoprint_mrbeamdoc/__init__.py

```diff
@@ -1,22 +1,191 @@
 #!/usr/bin/env python
 # coding=utf-8
+import os
+from enum import Enum
 
-import octoprint.plugin
+from __version import __version__
 
+_ROOT = os.path.abspath(os.path.dirname(__file__))
 
-from __version import __version__
 
+class MrBeamDocNotFoundException(Exception):
+    """Raised when the file requested is not found in the system"""
+
+    def __init__(self, mrbeamdoc_type, mrbeam_model, language):
+        super(MrBeamDocNotFoundException, self).__init__(
+            'The MrBeamDocType requested -> %(mrbeamdoc_type)s does not exist in the specified mrbeam_model -> %(mrbeam_model)s and language -> %(language)s' % {
+                'mrbeamdoc_type': mrbeamdoc_type, 'mrbeam_model': mrbeam_model, 'language': language})
+
+
+def get_doc_path(mrbeamdoc_type, mrbeam_model, language, extension='pdf'):
+    """
+    mrbeamdoc_type MrBeamDocType: Enum containing all the types of documents we are owning as Strings
+    mrbeam_model MrBeamModel: Enum containing all the existing mrbeam models as Strings
+    language String: language in 2 character format. e.g. 'de', 'en', ...
+    extension String: file format. e.g. 'pdf', ...
+
+    Return MrBeamDoc containing the needed paths to the requested file
+    raises MrBeamDocNotFoundException if the parameters are invalid or don't match any existing file
+    """
+
+    mrbeamdoc = MrBeamDocAvailable.get_mrbeamdoc_for(mrbeamdoc_type, mrbeam_model, language, extension)
+    if mrbeamdoc == None:
+        raise MrBeamDocNotFoundException(mrbeamdoc_type, mrbeam_model, language)
+
+    return mrbeamdoc
+
+
+def get_version():
+    return __version__
+
+
+class MrBeamDoc:
+    def __init__(self, definition, language, extension='pdf'):
+        self.definition = definition
+        self.language = language
+        self.extension = extension
+
+    def __repr__(self):
+        return 'MrBeamDoc(definition=%(definition)s, language=%(language)s, extension=%(extension)s)' % {
+            'definition': repr(self.definition),
+            'language': self.language.name,
+            'extension': self.extension,
+        }
+
+    def get_folder(self):
+        return os.path.join(_ROOT, 'docs')
```

Comment:
what will happen if 'docs' doesn't exist

Reply:
Nothing at this point since this just joins strings in a OS path context way AFAIK. But it would probably not be able to find any file at a later point since we rely on having them under docs... 🤔